riscv/insns/sret.h

```diff
@@ -16,11 +16,12 @@ s = set_field(s, MSTATUS_SPP, PRV_U);
 STATE.sstatus->write(s);
 p->set_privilege(prev_prv);
 if (!STATE.v) {
-  reg_t prev_virt = get_field(prev_hstatus, HSTATUS_SPV);
-  p->set_virt(prev_virt);
-
-  reg_t new_hstatus = set_field(prev_hstatus, HSTATUS_SPV, 0);
-  STATE.hstatus->write(new_hstatus);
+  if p->extension_enabled('H') {
+    reg_t prev_virt = get_field(prev_hstatus, HSTATUS_SPV);
+    p->set_virt(prev_virt);
+    reg_t new_hstatus = set_field(prev_hstatus, HSTATUS_SPV, 0);
+    STATE.hstatus->write(new_hstatus);
+  }
```

Comment:
I was a bit skeptical that `sret` would be the only place you'd need to make this change.  But based on some cursory grepping, it seems to be the case.

Reply:
same here.. grep resulted only in this one instruction requiring this change. Everywhere else its just a read operation.